pkg/chunk/chunk.go

```diff
@@ -327,3 +329,47 @@ func (c *Chunk) Samples(from, through model.Time) ([]model.SamplePair, error) {
 	interval := metric.Interval{OldestInclusive: from, NewestInclusive: through}
 	return prom_chunk.RangeValues(it, interval)
 }
+
+// Slice builds a new smaller chunk with data only from given time range
+func (c *Chunk) Slice(from, through model.Time) (*Chunk, error) {
+	if from > c.Through || through < c.From {
+		return nil, ErrSliceOutOfRange
+	}
+
+	itr := c.Data.NewIterator(nil)
```

Comment:
Sorry for the very dumb question. Can't we use `c.Data.Slice()`?

Reply:
It is certainly not a dumb question because I had discussed the same with Goutham :smile: 
First of all, it is not implemented for all encodings and secondly, for `BigChunk` It may overestimate. Also, Goutham mentioned that it is mainly used as an optimisation in the read path. See: https://github.com/cortexproject/cortex/blob/master/pkg/chunk/encoding/chunk.go#L53-L55